src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -512,6 +503,190 @@ public void TestVerbsProperty()
             Assert.DoesNotContain("closed", psi.Verbs, StringComparer.OrdinalIgnoreCase);
         }
 
+        [Theory]
+        [InlineData("")]
+        [InlineData("nofileextension")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Verbs_GetWithNoExtension_ReturnsEmpty(string fileName)
+        {
+            var info = new ProcessStartInfo { FileName = fileName };
+            Assert.Empty(info.Verbs);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Verbs_GetWithNoRegisteredExtension_ReturnsEmpty()
+        {
+            var info = new ProcessStartInfo { FileName = "file.nosuchextension" };
+            Assert.Empty(info.Verbs);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Verbs_GetWithNoEmptyStringKey_ReturnsEmpty()
+        {
+            const string Extension = ".noemptykeyextension";
+            const string FileName = "file" + Extension;
+
+            try
+            {
+                Registry.ClassesRoot.CreateSubKey(Extension);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // Skip this test if the user doesn't have permission to
+                // modify the registry.
+                return;
+            }
+
+            try
+            {
+                var info = new ProcessStartInfo { FileName = FileName };
+                Assert.Empty(info.Verbs);
+            }
+            finally
+            {
+                Registry.ClassesRoot.DeleteSubKeyTree(Extension);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Verbs_GetWithEmptyStringValue_ReturnsEmpty()
+        {
+            const string Extension = ".emptystringextension";
+            const string FileName = "file" + Extension;
+
+            try
+            {
+                RegistryKey key = Registry.ClassesRoot.CreateSubKey(Extension);
```

Comment:
Uh, that wouldn't delete the key of course. It would need something like [this](https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/IO/TempFile.cs) for a registry key.